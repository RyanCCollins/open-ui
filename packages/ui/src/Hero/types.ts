import * as React from 'react';
import Hero from './';

interface Props extends React.Props<typeof Hero> {
  backgroundColor?: string;
  background?: string;
  parallax?: boolean;
  size?: 'small' | 'medium' | 'large' | 'full';
}

export default Props;
