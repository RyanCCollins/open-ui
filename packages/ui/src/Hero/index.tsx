import * as React from 'react';
import Component from './styles';
import Props from './types';

export default function Hero({
  children,
  ...rest,
}: Props): JSX.Element {
  return (
    <Component {...rest}>
      {children}
    </Component>
  );
};
