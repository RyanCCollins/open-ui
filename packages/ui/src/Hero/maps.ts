export default {
  small: `${256 / 16}rem`,
  medium: `${512 / 16}rem`,
  large: `${1024 / 16}rem`,
  full: '100vh',
};
