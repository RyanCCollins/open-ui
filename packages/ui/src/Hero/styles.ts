const styled = require('styled-components').default;
const { css } = require('styled-components');
import sizeMap from './maps';

const backgroundCss = ({ background, backgroundColor }) => {
  if (!background) {
    return css`
      background: ${(props) => props.backgroundColor};
    `;
  }
  return css`
    background: url(${(props) => props.background}) no-repeat center center ${(props) => props.parallax ? 'fixed' : ''};
    background-size: cover;
  `;
};

export default styled.div`
  ${(props) => backgroundCss(props)}
  min-height: ${(props) => sizeMap[props.size || 'small']};
  text-align: center;
  width: 100%;
`;
