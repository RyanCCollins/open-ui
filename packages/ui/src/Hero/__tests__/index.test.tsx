import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Hero from '../';
import { heroProps } from './__mocks__/heroMocks.mock';

describe('<Hero /> rendering behavior', () => {
  it('should render the Hero component with default props', () => {
    const wrapper = shallow(
      <Hero>
        Some content
      </Hero>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with other props', () => {
    const wrapper = shallow(
      <Hero {...heroProps}>
        Some content
      </Hero>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with the provided prop values', () => {
    const wrapper = shallow(
      <Hero {...heroProps}>
        Some content
      </Hero>,
    );
    expect(shallowToJson(wrapper).props.backgroundColor).toEqual(heroProps.backgroundColor);
  });
});
