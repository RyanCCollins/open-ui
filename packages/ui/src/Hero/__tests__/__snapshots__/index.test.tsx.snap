// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> rendering behavior should render the Hero component with default props 1`] = `
<styled.div
  backgroundColor="#2d2d2d"
>
  Some content
</styled.div>
`;

exports[`<Hero /> rendering behavior should render with other props 1`] = `
<styled.div
  backgroundColor="ff6600"
>
  Some content
</styled.div>
`;
